### PR TITLE
Drop support for kubernetes < 1.20 for Azure extension provider

### PR DESCRIPTION
**Pull request by dimitar-kostadinov**
**How to categorize this PR?**

/area open-source
/kind cleanup
/platform azure

**What this PR does / why we need it**:
Drop support for kubernetes < 1.20 for Azure extension provider

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/6911

**Special notes for your reviewer**:

**Release note**:

```breaking operator
provider-azure no longer supports Shoots with Кubernetes version < 1.20.
```

### CODE REVIEW

**Review by MartinWeindel — APPROVED** [review]
/lgtm

File: charts/internal/cloud-provider-config/values.yaml [hunk 1]
```diff
@@ -1,4 +1,3 @@
-kubernetesVersion: 1.17.0
```
Comment (ialidzhikov) [inline charts/internal/cloud-provider-config/values.yaml:1]: I guess https://github.com/gardener/gardener-extension-provider-azure/blob/5fb2542f9ec1adf7f1b0cf98582a647b2fb82e49/pkg/controller/controlplane/valuesprovider.go#L442 can be removed now.

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**